andela/travel_tool_front
# #160386275 Fix travel duration

#### What does this PR do?
This PR ensure that the value of travel duration works as expected. For example, if the departure and return dates are the same days, then travel duration should be 1 and not 0 as it is currently. Also, for a one-way trip where there is no return date, this field should be 'Not Applicable'. 

#### Description of Task to be completed?
Given that a user submits a request having filled the same day(date) for `departure` and `return` dates. The value for the travel duration should be one(1) day.

#### How should this be manually tested?
- git clone https://github.com/andela/travel_tool_front.git
- git checkout bg-fix-travel-duration-160386275
- yarn install
- yarn start
- Login in the system and get token in cookies and copy
- Run the server under develop branch
- Using postman add trip using [this](https://gist.github.com/johnmutuma5/45cd66d793c52319bde97731be994ec5) and make sure you pass copied token in headers
- Add another trip, but for this case change trip to onWay
- Reload your browser to see the changes in my requests. 

#### Any background context you want to provide?
Postman is used in this case, since the feature for adding multi-trip and one way trip in form is under development and not merged to develop. 

#### What are the relevant pivotal tracker stories?

[#160386275](https://www.pivotaltracker.com/story/show/160386275)
#### Screenshots (if appropriate)
<img width="1402" alt="screen shot 2018-09-19 at 20 51 15" src="https://user-images.githubusercontent.com/32410518/45784282-ceff5880-bc70-11e8-8e85-858f185820b9.png">
<img width="1379" alt="screen shot 2018-09-20 at 15 58 44" src="https://user-images.githubusercontent.com/32410518/45819953-52ab5a80-bcee-11e8-9319-b98a8424b2c8.png">


#### Questions:
- Using 'Not Applicable' in the case of a one-way trip. Is this good? 
- I have seen that in the previous implementation, they have used arrival dates and departure dates and there are return dates. The calculation for travel duration is arrival dates and departure dates. My question is: If one schedule a return date as 12th but arrives back on date 13th, will 12th be used to calculate the travel duration and then the other to be overdue? If that is the case, where will be the column for all overdues, that is trips that go beyond the scheduled time? 

### NB
- If any of the information stated above is not clear, please, point out and I will clarify further.
